### PR TITLE
CFY-5135-ssl-on-file-server

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: cloudify_agent/api/pm/base.py [hunk 1]
```diff
@@ -250,6 +260,10 @@ def __init__(self, logger=None, **params):
         self.broker_pass = params.get('broker_pass', 'guest')
         self.host = params.get('host')
         self.deployment_id = params.get('deployment_id')
+        self.file_server_port = params.get(
```
Comment (dankilman) [inline cloudify_agent/api/pm/base.py]: remove

File: cloudify_agent/api/defaults.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
 REST_PORT = 80
 REST_PROTOCOL = 'http'
 SECURITY_ENABLED = False
+FILE_SERVER_PORT = 53229
```
Comment (dankilman) [inline cloudify_agent/api/defaults.py]: remove

File: cloudify_agent/installer/__init__.py [hunk 3]
```diff
@@ -183,12 +188,15 @@ def _create_agent_env(self):
...
             env.CLOUDIFY_BROKER_PORT: self.cloudify_agent.get('broker_port'),
+            env.CLOUDIFY_FILE_SERVER_PROTOCOL:
+                self.cloudify_agent.get('file_server_protocol'),
             env.CLOUDIFY_REST_PORT:
                 self.cloudify_agent.get('rest_port'),
             env.CLOUDIFY_REST_PROTOCOL: self.cloudify_agent.get(
                 'rest_protocol'),
             env.CLOUDIFY_SECURITY_ENABLED:
                 self.cloudify_agent.get('security_enabled'),
+            env.CLOUDIFY_REST_HOST: self.cloudify_agent.get('rest_host'),
```
Comment (dankilman) [inline cloudify_agent/installer/__init__.py]: remove

File: cloudify_agent/installer/__init__.py [hunk 4]
```diff
@@ -302,6 +310,8 @@ def download(self, url, destination=None):
             fh_num, destination = tempfile.mkstemp()
             os.close(fh_num)
 
+        # TODO: use requests instead, like we do with the rest-client
```
Comment (dankilman) [inline cloudify_agent/installer/__init__.py]: remove

File: cloudify_agent/installer/config/attributes.py [hunk 5]
```diff
@@ -37,10 +37,13 @@
     'protocol': {
         'group': 'connection'
     },
-    'ip': {
+    'file_server_host': {
```
Comment (dankilman) [inline cloudify_agent/installer/config/attributes.py]: fix

File: cloudify_agent/installer/runners/fabric_runner.py [hunk 6]
```diff
@@ -353,7 +353,8 @@ def mkdtemp(self, create=True, **attributes):
 
         return self.mktemp(create=create, directory=True, **attributes)
 
-    def download(self, url, output_path=None, **attributes):
+    def download(self, url, output_path=None, local_cert_file=None,
```
Comment (dankilman) [inline cloudify_agent/installer/runners/fabric_runner.py]: download_agent_package

File: cloudify_agent/installer/runners/fabric_runner.py [hunk 7]
```diff
@@ -379,14 +382,27 @@ def download(self, url, output_path=None, **attributes):
             self.logger.debug('Attempting to locate wget on the host '
                               'machine')
             self.run('which wget', **attributes)
-            command = 'wget -T 30 {0} -O {1}'.format(url, output_path)
+            # TODO: check args for https and cert use
+            if local_cert_file:
```
Comment (dankilman) [inline cloudify_agent/installer/runners/fabric_runner.py]: test matrix?

File: cloudify_agent/installer/runners/winrm_runner.py [hunk 8]
```diff
@@ -172,6 +172,9 @@ def download(self, url, output_path=None):
             output_path = self.mktemp()
 
         self.logger.info('Downloading {0}'.format(url))
+        # TODO: check args for https and cert use
```
Comment (dankilman) [inline cloudify_agent/installer/runners/winrm_runner.py]: in scope?

File: cloudify_agent/tests/api/pm/test_base.py [hunk 9]
```diff
@@ -33,6 +33,7 @@ def setUp(self):
             rest_host='127.0.0.1',
             broker_ip='127.0.0.1',
             file_server_host='127.0.0.1',
+            file_server_port='53229',
```
Comment (dankilman) [inline cloudify_agent/tests/api/pm/test_base.py]: use constant in all 53229 occurrences

File: cloudify_agent/shell/commands/daemons.py [hunk 10]
```diff
@@ -199,10 +209,10 @@ def create(**params):
     click.echo('Creating...')
 
     _create_rest_ssl_cert(attributes)
+    _create_broker_ssl_cert(attributes)
```
Comment (tehasdf) [inline cloudify_agent/shell/commands/daemons.py]: don't we want this to run after `get_broker_configuration`? `broker_ssl_cert` might be retrieved there

File: cloudify_agent/installer/__init__.py [hunk 11]
```diff
@@ -330,7 +335,15 @@ def create_custom_env_file_on_target(self, environment):
...
     def download(self, url, destination=None):
-        return self.runner.download(url, destination)
+        if self.cloudify_agent['windows']:
+            return self.runner.download(url, output_path=destination)
+        else:
+            local_cert_file = self.cloudify_agent.get('local_rest_cert_file')
+            verify_certificate = True if local_cert_file else False
+            return self.runner.download(url,
+                                        output_path=destination,
+                                        verify_certificate=verify_certificate,
```
Comment (dankilman) [inline cloudify_agent/installer/__init__.py]: `bool(local_cert_file)`

File: cloudify_agent/installer/runners/fabric_runner.py [hunk 12]
```diff
@@ -353,7 +353,8 @@ def mkdtemp(self, create=True, **attributes):
 
         return self.mktemp(create=create, directory=True, **attributes)
 
-    def download(self, url, output_path=None, **attributes):
+    def download(self, url, output_path=None, verify_certificate=True,
```
Comment (dankilman) [inline cloudify_agent/installer/runners/fabric_runner.py]: skip_verification

File: cloudify_agent/tests/installer/config/test_configuration.py [hunk 13]
```diff
@@ -125,6 +125,8 @@ def test_prepare_secured(self):
             'rest_port': '443',
             'rest_protocol': 'https',
             'file_server_host': 'localhost',
+            'file_server_port': 53229,
```
Comment (dankilman) [inline cloudify_agent/tests/installer/config/test_configuration.py]: clean all 53229

**Pull request merged**